### DISCUSSION TITLE
Up-to-date kata rank definitions?

### DISCUSSION THREAD

**Discussion by MIC132** [codewars/codewars.com#2924] [Q&A]
This page:
https://docs.codewars.com/curation/references/kata-ranks
says the definitions there are outdated.

Where would I find up-to-date ones?

**Comment by hobovsky** [comment]
Updated definitions of rank are not available at the moment, because there's no real agreement on what particular ranks would mean. Different users perceive difficulty differently, and it's not easy to derive some common definitions from this.

There was one attempt of updating the descriptions, which can be seen [here](https://deploy-preview-461--reverent-edison-2864ea.netlify.app/curation/references/kata-ranks/). But currently it's only a draft, not an officially approved version, but I personally like the proposal and I think it reflects current state quite well.

**Reply by MIC132** [reply]
So if I were planning on making a kata, I would basically need to go with my gut?

**Reply by Madjosz** [reply]
Yes. Estimated rank is only shown at the kata search. The rank is determined by the approver in the end and they should consider the votes of the solvers for that.

**Reply by hobovsky** [reply]
No, not really. There are no hard guidelines, but there is still a few possibilities:
- Use analogy to existing, approved kata, if possible and if some common idea on their difficulty can be established, or
- Leave out the rank suggestion empty, because beta testers will vote anyway
